lib/train/transports/ssh_connection.rb

```diff
@@ -52,6 +52,7 @@ def initialize(options)
       @bastion_host           = @options.delete(:bastion_host)
       @bastion_user           = @options.delete(:bastion_user)
       @bastion_port           = @options.delete(:bastion_port)
+      check_config_options
```

Comment:
I think this needs to happen first, not last. If a user provides a username (or port) in the connection options as well as in the SSH config files, we want the provided option to take precedence. As written, the SSH config file will overwrite the user-provided option, which is not what we want.

Reply:
Hi @clintoncwolfe, i have tried earlier to use it at first, but its not working as we are expecting, when we are not providing the user, SSH config file is not overwriting it, as we are setting default user value to 'root' https://github.com/inspec/train/blob/ef35cec62de6f0ace9361e2d2fe73169d7878332/lib/train/transports/ssh.rb#L47

Reply:
We can't have that precedence - it needs to be CLI options, config file, default. Adding your change will make it config file, CLI options, default. Consider adding code above the call to `super(options)` to detect if port and username are being set and if so don't override them in read_options_from_ssh_config.